tests/integration/targets/vmware_ops_provision_virtual_esxi_test/tasks/main.yml

```diff
@@ -0,0 +1,83 @@
+---
+- name: Test Virtual Esxi Provisioning on Simulator
+  tags: integration-ci
+  ansible.builtin.include_role:
+    name: cloud.vmware_ops.provision_virtual_esxi
+
+- name: Test Virtual Esxi Provisioning on vCenter Env
+  tags: eco-vcenter-ci
+  block:
+  - name: Add Resource Pool
+    community.vmware.vmware_resource_pool:
+      hostname: "{{ provision_virtual_esxi_hostname }}"
+      username: "{{ provision_virtual_esxi_username }}"
+      password: "{{ provision_virtual_esxi_password }}"
+      datacenter: "{{ provision_virtual_esxi_datacenter }}"
+      cluster: "{{ provision_virtual_esxi_cluster }}"
+      validate_certs: "{{ provision_virtual_esxi_validate_certs }}"
+      resource_pool: "{{ resource_pool_name }}"
+      state: present
+
+  - name: Provision Virtual Esxi
+    ansible.builtin.import_role:
+      name: cloud.vmware_ops.provision_virtual_esxi
+    vars:
+      provision_virtual_esxi_resource_pool: "{{ resource_pool_name }}"
+
+  - name: Wait until ESXi VM is powered off after OS installation
+    community.vmware.vmware_guest_info:
+      hostname: "{{ provision_virtual_esxi_hostname }}"
+      username: "{{ provision_virtual_esxi_username }}"
+      password: "{{ provision_virtual_esxi_password }}"
+      port: "{{ provision_virtual_esxi_port }}"
+      datacenter: "{{ provision_virtual_esxi_datacenter }}"
+      validate_certs: "{{ provision_virtual_esxi_validate_certs }}"
+      name: "{{ provision_virtual_esxi_vms[0].name }}"
+    register: _esxi_power_status
+    until: _esxi_power_status.instance.hw_power_status == 'poweredOff'
+    retries: 30
+    delay: 20
```

Comment:
The [role](https://github.com/redhat-cop/cloud.vmware_ops/blob/main/roles/provision_virtual_esxi/tasks/deploy_new_esxi_hosts.yml#L57C1-L77C12) already waits until the host is powered off OR the IP is assigned, why do we need to wait?

Reply:
The premise/policy of vmware is that the vm will went off at the end of an esxi installation - but since the success of the role is determine even if the IP is catched before the vm powered off, the other tasks will continue and will fail. the failure will occur when suddenly the machine will be turned off at the middle of the task.
so this task wait for some time in case the IP is catched before the vm is powered off.